nanoFramework.Device.OneWire/OneWireMaster.cs

```diff
@@ -10,17 +10,52 @@
 namespace nanoFramework.Device.OneWire
 {
     /// <summary>
-    /// Represents a 1-Wire bus controller. The class provides methods and properties that an application can use to interact with the 1-Wire bus and connected devices.
+    /// Initializes a new instance of the <see cref="OneWireMaster"/> class.
+    /// Represents a 1-Wire bus master. The class provides methods and properties that an application can use to interact with the 1-Wire bus and connected devices.
     /// </summary>
-    public class OneWireController
+    public class OneWireMaster : IDisposable
     {
+        // flag to signal that an instance of the class has been created
+        private static bool s_opened = false;
+
         // this is used as the lock object 
-        // a lock is required because multiple threads can access the SerialDevice
-        private object _syncLock = new object();
+        // a lock is required because multiple threads can access the FindAllDevices method
+        private readonly object _syncLock;
 
         // this is the backing field for the serial number on discovery or when performing a read/write operation
         // need to create it here to be used in native
-        private byte[] _serialNumber = new byte[8];
+        private byte[] _serialNumber;
+
+        private bool _disposedValue;
+
+        // external One Wire functions from link layer owllu.c
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OneWireMaster"/> class.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">If there is already another instance of the <see cref="OneWireMaster"/> class.</exception>
+        public OneWireMaster()
+        {
+            if (!s_opened)
+            {
+                _disposedValue = false;
```

Comment:
Maybe just disposed? I feel adding value is confusing

Reply:
It's what we get on VS template for Dispose implementation 😅  but will change, no sweat.